service/history/timer_queue_standby_task_executor.go

```diff
@@ -517,11 +609,6 @@ func (t *timerQueueStandbyTaskExecutor) processTimer(
 		return nil
 	}
 
-	if !mutableState.IsWorkflowExecutionRunning() {
-		// workflow already finished, no need to process the timer
-		return nil
-	}
```

Comment:
State machine timer task needs to be processed even if the workflow is not running. So, I moved this check to each of the other timer tasks `actionFn`.